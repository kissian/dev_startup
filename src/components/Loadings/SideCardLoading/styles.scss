@mixin nth-children($points...) {
  @each $point in $points {
      &:nth-child(#{$point}) {
          @content;
      }
  }
}

.SideCardLoding {
  clear: both;
  content: "";
  display: flex;
  flex-direction: column;
  border: 1px solid #eee;
  border-radius: 0;
  width: 100%;
  margin-bottom: 30px;
  .loadingBox {
    width: 100%;
    height: 50px;
    animation: pulse 1s infinite ease-in-out;
  }
  .loadingLine {
    height: 15px;
    margin: 10px 0 10px 10px;
    width: 30%;
    animation: pulse 1s infinite ease-in-out;
  }
}

@keyframes pulse {
  0% {
      background-color: rgba(165, 165, 165, 0.1)
  }
  50% {
      background-color: rgba(165, 165, 165, 0.3)
  }
  100% {
      background-color: rgba(165, 165, 165, 0.1)
  }
}