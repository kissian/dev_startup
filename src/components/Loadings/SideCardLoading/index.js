import React from 'react';
import {
    Row, Col,
} from 'react-bootstrap';
import './styles.scss';

const SideCardLoading = () => (
    <Row className="SideCardLoding">
        <div className="loadingLine" />
        <div>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
            <Col xs={6} md={4}>
                <div className="thumbnail loadingBox" />
            </Col>
        </div>
    </Row>
);

export default SideCardLoading;
