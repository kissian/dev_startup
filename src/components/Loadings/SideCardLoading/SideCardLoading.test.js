import React from 'react';
import { shallow } from 'enzyme';
import SideCardLoading from './index';

describe('SideCardLoading component', () => {
    let component = null;
    it('renders component', () => {
        component = shallow(<SideCardLoading />);
    });
    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});
