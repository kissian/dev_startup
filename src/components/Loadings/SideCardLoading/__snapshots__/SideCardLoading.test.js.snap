// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideCardLoading component matches snapshot 1`] = `
<Row
  bsClass="row"
  className="SideCardLoding"
  componentClass="div"
>
  <div
    className="loadingLine"
  />
  <div>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      xs={6}
    >
      <div
        className="thumbnail loadingBox"
      />
    </Col>
  </div>
</Row>
`;
